### PR TITLE
plugins/lsp/pylsp: Add python option to generate obtain the pylsp from.

### PR DESCRIPTION
**Pull request by purepani**
Currently, pkgs.python3Packges is used by default and is unable to be changed. Without this, using features from newer python versions in mypy(and maybe other plugins) is impossible since mypy uses it's own python interpreter that it's installed under for getting typing hints afaik.
Fixes https://github.com/nix-community/nixvim/issues/1456

### CODE REVIEW

File: plugins/lsp/language-servers/pylsp.nix [hunk 1]
```diff
@@ -12,7 +12,11 @@ in
 {
   # All settings are documented here:
   # https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md
-
+  options.plugins.lsp.servers.pylsp.python = mkOption {
```
Comment (GaetanLepage) [inline plugins/lsp/language-servers/pylsp.nix:15]:
```suggestion
  options.plugins.lsp.servers.pylsp.pythonPackage = mkOption {
```
This is a clearer name IMO. cc @MattSturgeon

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
This is very useful indeed !

File: plugins/lsp/language-servers/pylsp.nix (hunk 1 above)
Comment (MattSturgeon) [reply plugins/lsp/language-servers/pylsp.nix:15]: Agreed. Defining `options.plugins.lsp.servers.pylsp = {` once would also be an improvement imo, but that's a nit.

Comment (purepani) [reply plugins/lsp/language-servers/pylsp.nix]: Not sure how to make the diff be smaller, but both of these are implemented.

File: plugins/lsp/language-servers/pylsp.nix [hunk 2]
```diff
@@ -12,483 +12,490 @@ in
 {
   # All settings are documented here:
   # https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md
```
Comment (MattSturgeon) [inline plugins/lsp/language-servers/pylsp.nix:14]: Can you move these comments down, to above the `settings = {` line?

File: plugins/lsp/language-servers/pylsp.nix (hunk 1 above)
Comment (MattSturgeon) [reply plugins/lsp/language-servers/pylsp.nix]: > Not sure how to make the diff be smaller

Can't be helped due to the auto-formatting.

You can view the diff with whitespace ignored though: https://github.com/nix-community/nixvim/pull/1785/files?w=1

File: plugins/lsp/language-servers/pylsp.nix (hunk 2 above)
Comment (purepani) [reply plugins/lsp/language-servers/pylsp.nix]: Done!

**Review by MattSturgeon — APPROVED** [review]
LGTM, thanks!

**Comment by GaetanLepage** [comment]
@Mergifyio queue

**Pull request merged**